desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java

```diff
@@ -751,7 +751,7 @@ protected boolean onWindowClosing() {
 
                 if (unsavedChangesChoice == 0) { // save changes
                     _saveButton.doClick();
-                } else if (unsavedChangesChoice == 2) { // cancel closing
+                } else if (unsavedChangesChoice != 1) { // cancel closing
```

Comment:
Yeah, I know that, but I still don't agree.

We match the indexes because we chose to match the indexes. Keeping the code less readable because of an implementation detail makes no sense to me.

I'd even argue that there's a good change that the use of numbers instead of constants helped this bug creep in in the first place.